source/dem/dem.cc

```diff
@@ -322,27 +351,107 @@ DEMSolver<dim>::setup_background_dofs()
 }
 
 template <int dim>
-bool
+void
 DEMSolver<dim>::load_balance()
 {
-  if (simulation_control->get_step_number() % repartition_frequency == 0)
-    {
-      pcout << "-->Repartitionning triangulation" << std::endl;
-      triangulation.repartition();
+  pcout << "-->Repartitionning triangulation" << std::endl;
+  triangulation.repartition();
 
-      cells_local_neighbor_list.clear();
-      cells_ghost_neighbor_list.clear();
+  cells_local_neighbor_list.clear();
+  cells_ghost_neighbor_list.clear();
 
-      cell_neighbors_object.find_cell_neighbors(triangulation,
-                                                cells_local_neighbor_list,
-                                                cells_ghost_neighbor_list);
+  cell_neighbors_object.find_cell_neighbors(triangulation,
+                                            cells_local_neighbor_list,
+                                            cells_ghost_neighbor_list);
 
-      boundary_cell_object.build(triangulation, parameters.floating_walls);
-      return true;
-    }
+  boundary_cell_object.build(triangulation, parameters.floating_walls);
+
+  const unsigned int maximum_particle_number_on_proc =
+    Utilities::MPI::max(particle_handler.n_locally_owned_particles(),
+                        mpi_communicator);
+  const unsigned int minimum_particle_number_on_proc =
+    Utilities::MPI::min(particle_handler.n_locally_owned_particles(),
+                        mpi_communicator);
+  const unsigned int maximum_cell_number_on_proc =
+    Utilities::MPI::max(triangulation.n_active_cells(), mpi_communicator);
+  const unsigned int minimum_cell_number_on_proc =
+    Utilities::MPI::min(triangulation.n_active_cells(), mpi_communicator);
```

Comment:
Instead of reporting min and max, I would report Min - Max and Avg.
You can use the following function :+1: 
https://www.dealii.org/current/doxygen/deal.II/structUtilities_1_1MPI_1_1MinMaxAvg.html
This way in a single MPI call you get all the statistics and you can report them.

Reply:
Done!